web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -0,0 +1,370 @@
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Box, ButtonSecondary, Link, Text } from 'design';
+import * as Icons from 'design/Icon';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { requiredField } from 'shared/components/Validation/rules';
+import useAttempt from 'shared/hooks/useAttemptNext';
+
+import { TextSelectCopyMulti } from 'teleport/components/TextSelectCopy';
+import { usePingTeleport } from 'teleport/Discover/Shared/PingTeleportContext';
+import {
+  HintBox,
+  SuccessBox,
+  WaitingInfo,
+} from 'teleport/Discover/Shared/HintBox';
+import {
+  AwsOidcDeployServiceResponse,
+  integrationService,
+} from 'teleport/services/integrations';
+import { useDiscover, DbMeta } from 'teleport/Discover/useDiscover';
+import { DiscoverEventStatus } from 'teleport/services/userEvent';
+
+import {
+  ActionButtons,
+  HeaderSubtitle,
+  TextIcon,
+  useShowHint,
+  Header,
+  DiscoverLabel,
+  AlternateInstructionButton,
+  Mark,
+} from '../../../Shared';
+
+import { DeployServiceProp } from '../DeployService';
+import { hasMatchingLabels, Labels } from '../../common';
+
+import type { Database } from 'teleport/services/databases';
+
+export function AutoDeploy({ toggleDeployMethod }: DeployServiceProp) {
+  const { emitErrorEvent, nextStep, emitEvent, agentMeta, updateAgentMeta } =
+    useDiscover();
+  const { attempt, setAttempt } = useAttempt('');
+  const [showLabelMatchErr, setShowLabelMatchErr] = useState(true);
+
+  const [taskRoleArn, setTaskRoleArn] = useState('TeleportDatabaseAccess');
+  const [deploySvcResp, setDeploySvcResp] =
+    useState<AwsOidcDeployServiceResponse>();
+  const [deployFinished, setDeployFinished] = useState(false);
+
+  const hasDbLabels = agentMeta?.agentMatcherLabels?.length;
+  const dbLabels = hasDbLabels ? agentMeta.agentMatcherLabels : [];
+  const [labels, setLabels] = useState<DiscoverLabel[]>([
+    { name: '*', value: '*', isFixed: dbLabels.length === 0 },
+  ]);
+  const agent = agentMeta as DbMeta;
+
+  useEffect(() => {
+    // Turn off error once user changes labels.
+    if (showLabelMatchErr) {
+      setShowLabelMatchErr(false);
+    }
+  }, [labels]);
+
+  function handleDeploy(validator: Validator) {
+    if (!validator.validate()) {
+      return;
+    }
+
+    if (!hasMatchingLabels(dbLabels, labels)) {
+      setShowLabelMatchErr(true);
+      return;
+    }
+
+    setShowLabelMatchErr(false);
+    setAttempt({ status: 'processing' });
+    integrationService
+      .deployAwsOidcService(agent.integrationName, {
+        deploymentMode: 'database-service',
+        region: agent.selectedAwsRdsDb?.region,
+        subnetIds: agent.selectedAwsRdsDb?.subnets,
+        taskRoleArn,
+        databaseAgentMatcherLabels: labels,
+      })
+      .then(setDeploySvcResp)
```

Comment:
Should this set the attempt to success?

Reply:
no it wasn't necessary (save a re-render?) because right after this call succeeds, we move to `pinging` for the db service to pick up the database (which is still a processing state)

i left a clarifying comment